sklearn/linear_model/ridge.py

```diff
@@ -327,7 +327,7 @@ def ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
         y = check_array(y, dtype=np.float64, ensure_2d=False, order='F')
     else:
         X = check_array(X, accept_sparse=['csr', 'csc', 'coo'],
-                        dtype=np.float64)
+                        dtype='numeric')
```

Comment:
> Wouldn't sticking with np.float64 cancel the second bullet point?

Yes it would, my point is that we don't have a test to prove it. I am not sure it is easy to test it though.

>  I test the code is by adding the following lines at the end of my local version

You can use `nosetests sklearn/linear_model/tests/test_ridge.py -v` or `pytest sklearn/linear_model/tests/test_ridge.py -v`

To run one particular test, you can use `nosetests sklearn/linear_model/tests/test_ridge.py:test_dtype_match` or `pytest sklearn/linear_model/tests/test_ridge.py::test_dtype_match`